packages/flutter_tools/lib/src/commands/build_aar.dart

```diff
@@ -85,17 +86,20 @@ class BuildAarCommand extends BuildSubCommand {
   @override
   Future<FlutterCommandResult> runCommand() async {
     final Set<AndroidBuildInfo> androidBuildInfo = <AndroidBuildInfo>{};
-    final Iterable<AndroidArch> targetArchitectures = stringsArg('target-platform')
-      .map<AndroidArch>(getAndroidArchForName);
+
+    final Iterable<AndroidArch> targetArchitectures =
+        stringsArg('target-platform').map<AndroidArch>(getAndroidArchForName);
+
+    final String buildNumber = argParser.options.containsKey('build-number')
+      ? stringArg('build-number')
```

Comment:
Should this also check that the `build-number` is in a valid format?

Reply:
@blasten I chose not to validate it because the other usage in the tools codebase is also not validated. you can see this in the `getBuildInfo` method in packages/flutter_tools/lib/src/runner/flutter_command.dart